Feature/#438 update grid

jomurgel opened the pull request:
Closes #438

### DESCRIPTION ###
Much of the documentation for these updates were added to the documentation doc below: https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#heading=h.b1ma2w6tlbd

**The main push here was removing Neat entirely and rebuilding the grid entirely with flexbox from top to bottom.**

**These grid helpers and mixins showed in the doc above removes the unnecessary flexibility of Neat 1.9x and provides the ability to create the following grid/layout options.**
- **Full Width**
- **Container Width**
- **50%**
- **33%**
- **25%**
- **~66% — Less Gutter accommodation.**

### Updates
- Added .vscode to .gitignore.
- Replaces any instance of the commonly-used rem(24) with a new $space variable.
- Added general style tweaks for IE11 or general accommodations for the new grid.
- Button styles to be more cross-browser compatible.
- Block naming conventions (below).
- Updated scaffolding to use any of the added mixins or classnames.
- Cleaned up hamburger open/close button.
- Fixed search form container elements.
- Added function 
- Replace normalize with Sanitize — seen in #442 
- Inlines Instances of Triangle Mixin with inline styles.

![screenshot 2018-12-17 11 18 36](https://user-images.githubusercontent.com/5230729/50106607-8b73a680-01ed-11e9-8f4b-42ffd340c4aa.jpg)

### Removes
- Bourbon as a dependency
- Bourbon from Gulpfile.js
- Hyphens Mixin
- Font Legibility Mixin
- Margin Auto Mixin
- Margin Padding Reset Mixin
- Vertical Align Mixin
- Word Break Mixin
- Z-Index Mixin
- Need for Clearfix Mixin, removed mixin, and styled the areas that used it to use flexbox.

### Added
- New template-sidebar-left.php template.
- Empty style partial for the generic content block.
- Default styles for blockquote to match typical output (extends h3).
- Added function to push YOAST below all content to hooks.php
- Added function to escape content output the way that WP does it — replacing any instances of force_balance_tags() which doesn’t escape anything.
- Directional Property Mixin — helper.
- Margin Mixin
- Padding Mixin
- Position Mixin
- Shorthand Function

I made several opinionated additions here that we're constantly making updates during projects. Those are listed here: https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#heading=h.b1ma2w6tlbd

I still think there are many areas we could improve the mixin usage and markup, but there are other issues for that.

### SCREENSHOTS ###
**ACF Blocks**
![screencapture-192-168-1-237-3000-acf-blocks-2018-12-17-11_10_57](https://user-images.githubusercontent.com/5230729/50106221-72b6c100-01ec-11e9-9d98-a92f9b2551f5.png)

**Quarter Grid**
![screenshot 2018-12-17 11 13 35](https://user-images.githubusercontent.com/5230729/50106328-ce814a00-01ec-11e9-9e47-443f1dc38fdc.jpg)

**Radio/Checkbox/Select Updates**
![screen shot 2018-12-17 at 10 54 25 am](https://user-images.githubusercontent.com/5230729/50106396-fbcdf800-01ec-11e9-87cd-53d587e7a1c0.jpg)

### OTHER ###
- [x] Is this issue accessible? (Section 508/WCAG 2.0AA)
- [x] Does this issue pass all the linting? (PHPCS, ESLint, SassLint)
- [x] Does this pass CBT?

### DOCUMENTATION ###
The doc below has updates and documentation for these changes and would, if approved, be added to the Wiki: 
https://docs.google.com/document/d/1o1zfqeT5bcdu4VsCHCOKCbO72YMKjb7509I83srkzKU/edit#

jomurgel committed:
removes neat from style.scss and gulpfile.js

jomurgel committed:
adds media query mixin

jomurgel committed:
removes space

jomurgel committed:
imports media query plugin

jomurgel committed:
combines necessary grid variables and breakpoint variables into one partial

jomurgel committed:
removes rows partial

jomurgel committed:
updates media query usages to match new convention

jomurgel committed:
removes grid helpers

jomurgel committed:
remove instances of `col-` classes

jomurgel committed:
removes container wrappers and appends class and id to body

jomurgel committed:
removes wrapping primary container class

jomurgel committed:
adds container width grid var

jomurgel committed:
removes homepage partial in favor of template-sidebar-right view

jomurgel committed:
changes gutter var to 20

jomurgel committed:
Cleans up layout structure to utilize flexbox and easy-to-use full/container width elemenets

jomurgel committed:
moves sidebar elements to template partial

jomurgel committed:
updates margin output to utilize mixin

jomurgel committed:
replaces marign with margin mixin

jomurgel committed:
replaces padding with padding mixin

jomurgel committed:
rebuilds header with flexbox and removes span elements from neat

jomurgel committed:
replaces testing.test with wds.test

jomurgel committed:
adds wrapper for flexbox with sidebar

jomurgel committed:
adds container class to articles

jomurgel committed:
adds container class to sidebar

jomurgel committed:
update body class/id and adds grid-container class to header to init flexbox

jomurgel committed:
adds grid wrapper to single post temlate

jomurgel committed:
builds

jomurgel committed:
removes grid-x class (invalidated)

jomurgel committed:
adds filter to move yoast below all elements

jomurgel committed:
adds helper class and removes global acf margins

jomurgel committed:
adds margins to site-main class

jomurgel committed:
wraps sidebar right template

jomurgel committed:
wraps single post (sidebar)

jomurgel committed:
updates header wrap

jomurgel committed:
file cleanup

jomurgel committed:
changes grid-wrapper to display-flex helper

jomurgel committed:
updates button padding to place nice with inputs

jomurgel committed:
cleanup

jomurgel committed:
removes button style restrictions from header button action

jomurgel committed:
adds spacer variable

jomurgel committed:
adds padding to has-background elements

jomurgel committed:
Replaces any instance of rem(24) with $space variable

jomurgel committed:
adds block width field to other options

jomurgel committed:
adds vscode to gitignore

jomurgel committed:
adds container width to container options

jomurgel committed:
adds margin auto and width to site-main

jomurgel committed:
update container width output

jomurgel committed:
updates 'cell' cllass with 'container' class

jomurgel committed:
makes sure container is auto left/right margin

jomurgel committed:
removes unused classes

jomurgel committed:
adds space between accordion elements

jomurgel committed:
updates cta block to follow new conventions

jomurgel committed:
sets to full width and cleans up markup

jomurgel committed:
updates headings for cta

jomurgel committed:
adds helpers for jutify center and align center

jomurgel committed:
moves content width to display options tab

jomurgel committed:
removes undnerlines from buttons

jomurgel committed:
ads font weight

jomurgel committed:
adjusts carouel for new grid

jomurgel committed:
adds empty and missing general content partial

jomurgel committed:
renames _template-sidebar-right to _has-sidebar.scss

jomurgel committed:
removes margin

jomurgel committed:
removes padding

jomurgel committed:
sets postion for background figure

jomurgel committed:
Updates block naming conventions

jomurgel committed:
adds display options for width

jomurgel committed:
updates selector

jomurgel committed:
builds js

jomurgel committed:
brings button styles up to date with links

jomurgel committed:
avoid link conflicts with links with button class

jomurgel committed:
adds helpers for third and half grid elements

jomurgel committed:
adds mixin partial for grid elements

jomurgel committed:
empties partials

jomurgel committed:
updates markup to utilize new grid

jomurgel committed:
builds

jomurgel committed:
resolves phpcs issues

jomurgel committed:
adds filter to run content through — replace force balance tags

jomurgel committed:
adds missing return statement in docblock

jomurgel committed:
styles search from

jomurgel committed:
removes margin from hentry elements

jomurgel committed:
hides pagination of no pages exist

jomurgel committed:
builds

jomurgel committed:
adds missing return doc

jomurgel committed:
adds new sidebar left template

jomurgel committed:
updates documentation and adds helper class

jomurgel committed:
adds classname depending on template in use

jomurgel committed:
removes has-sidebar template and reverts to empty homepage template

jomurgel committed:
adds tempalte sidebar left template to hooks for has-sidebar class

jomurgel committed:
adds helper classes for sidebar elements

jomurgel committed:
adds mixins for sidebar elements

jomurgel committed:
builds

jomurgel committed:
removes margin from last paragraph

jomurgel committed:
adds documentation for grid classes

jomurgel committed:
adds quarter mixin

jomurgel committed:
fixes ul list

jomurgel committed:
fixes blockquote margin

jomurgel committed:
styles blockquote defaults

jomurgel committed:
adds defaults for typography

jomurgel committed:
builds

jomurgel committed:
adds color classes to style backgrounds

jomurgel committed:
wraps heading to allow for full-width elements

jomurgel committed:
adds max-width and margin auto left/right on menu area

jomurgel committed:
styles top of list instead of bottom

jomurgel committed:
adds padding to footer

jomurgel committed:
builds

jomurgel committed:
updates scaffolding styles/elements to follow new conventions

jomurgel committed:
builds

jomurgel committed:
adds default styles for radio/checkbox and select dropdown

jomurgel committed:
updates docs

jomurgel committed:
resolves alignment issues in main navigation

jomurgel committed:
resolves close icon inconsistancies

jomurgel committed:
builds

jomurgel committed:
adds fixes for ie11

jomurgel committed:
builds

jomurgel committed:
adds container width display options to carousel

jomurgel committed:
removes full-width mixin

jomurgel committed:
adds full-width mixin for helper

jomurgel committed:
updates media query

jomurgel committed:
builds

jomurgel committed:
sets search form container input min-width to 0

jomurgel committed:
builds

jomurgel committed:
removes omega reset

jomurgel committed:
uses color variables in form elements for radio/checkbox

jomurgel committed:
determins that yoast is active before running metabox_prio filter

jomurgel committed:
replaces normalize with sanitize

jomurgel committed:
removes bourbon as a dependency

jomurgel committed:
removes bourbon from gulp

jomurgel committed:
replaces " with ' in gulpfile

jomurgel committed:
adds directional property mixin

jomurgel committed:
adds margin mixin

jomurgel committed:
adds padding mixin

jomurgel committed:
adds position mixin

jomurgel committed:
adds size mixin

jomurgel committed:
adds unpack shorthand mixin

jomurgel committed:
adds new mixins

jomurgel committed:
removes clearings partial

jomurgel committed:
adds mixin utilities into subfilder

jomurgel committed:
updates clearfix

jomurgel committed:
replaces triangle mixin with markup

jomurgel committed:
removes font smoothing and legibility — set by default

jomurgel committed:
removes wordbreak mixin and inlines requirement — used once

jomurgel committed:
removes z-index mixin — not used

jomurgel committed:
removes vertical alignment mixin

jomurgel committed:
remove margin-auto mixin and inlines — used once

jomurgel committed:
removes hyphens mixin — unsued

jomurgel committed:
removes margin-padding reset and inlines

jomurgel committed:
cleanup mixins and utilities

jomurgel committed:
dds is size, is length and contains falsy function validators

jomurgel committed:
resolves issues with size mixin

jomurgel committed:
removes clearings styles

jomurgel committed:
updates comment odd background color

jomurgel committed:
removes clearfix

jomurgel committed:
removes instances of clearfix or floats

jomurgel committed:
adds flexbox to comments

jomurgel committed:
adds sidebar class to single posts template

jomurgel committed:
cleans up post navigation

jomurgel committed:
adds empty style containers for pagination container

jomurgel committed:
builds

jomurgel committed:
resolve link block issues in social nav

jomurgel committed:
adds container to post navigation

jomurgel committed:
fixes styles issues with header action button/form container

jomurgel committed:
builds

jomurgel committed:
removes unused description

jomurgel committed:
resolves html validation errors

jomurgel committed:
adds missing role="main" to <main> tag

coreymcollins commented on 404.php:
```diff
@@ -9,24 +9,22 @@
+			<div class="page-content">
 
-					<p><?php esc_html_e( 'It seems we can’t find what you’re looking for. Perhaps searching can help.', '_s' ); ?></p>
+				<p><?php esc_html_e( 'It seems we can’t find what you’re looking for. Perhaps searching can help.', '_s' ); ?></p>
```
Since we're not escaping the apostrophe in `can't` here, I'm guessing it's rich text and should be replaced & escaped to match the heading on line 16.

coreymcollins commented on archive.php:
```diff
@@ -9,8 +9,7 @@
 
-	<div class="primary content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main" role="main">
```
`role` is invalid here and shouldn't be used: https://www.w3.org/TR/html-aria/#document-conformance-requirements-for-use-of-aria-attributes-in-html

jomurgel replied:
Ah, thanks. I thought that was a thing. I know the older WP core themes doubled up that way so I erred on the side of caution.

coreymcollins commented on assets/sass/base/_scaffolding.scss:
```diff
@@ -6,7 +6,6 @@
 // IE11 Tweak
 //-----------------------------------------
 .page-template-template-scaffolding .primary {
```
We don't use the `primary` class at all anymore with this update, do we?

jomurgel replied:
Ah, no. This should have been removed.

coreymcollins commented on assets/sass/base/_typography.scss:
```diff
@@ -19,46 +24,56 @@ textarea {
 // Global Header Resets.
 @for $i from 1 through 6 {
 	h#{$i} {
```
Can we also add `.h#{$i}` here so we can capture not only the actual headings but the heading classes like `.h1`, `.h2`, etc?

coreymcollins commented on assets/sass/modules/content-blocks/_fifty-fifty.scss:
```diff
@@ -3,10 +3,4 @@
 //--------------------------------------------------------------
```
If this is empty now, let's just remove the partial. It can always be added if we need to adjust styles, but I'd rather avoid having an empty file just for the sake of having it.

coreymcollins commented on assets/sass/modules/content-blocks/_general.scss:
```diff
@@ -0,0 +1,7 @@
+//--------------------------------------------------------------
```
Same note as on fifty-fifty – if this is empty, let's just remove it.

coreymcollins commented on assets/sass/modules/content-blocks/_global.scss:
```diff
@@ -13,6 +13,10 @@
 	} // .content-block-title
 
+	&.has-background {
+		@include padding($gutter * 2 null);
```
I think this should also include left and right padding. If something is container-width and has a background set, the content runs right up to the edge.

coreymcollins commented on assets/sass/modules/content-blocks/_recent-posts.scss:
```diff
@@ -3,18 +3,4 @@
 //--------------------------------------------------------------
```
If empty, let's remove.

coreymcollins commented on assets/sass/modules/content-blocks/_related-posts.scss:
```diff
@@ -3,18 +3,4 @@
 //--------------------------------------------------------------
```
If empty, let's remove.

coreymcollins commented on assets/sass/modules/navigation/_post-navigation.scss:
```diff
@@ -4,21 +4,45 @@
 .post-navigation {
+
+	.nav-links {
+		@extend .display-flex;
```
I'd like to avoid using `@extend` where possible as it can cause headaches. Is there any reason to not add `display-flex` as a class to this element?

coreymcollins commented on assets/sass/modules/navigation/_post-navigation.scss:
```diff
@@ -4,21 +4,45 @@
+			font-weight: 700;
+		} // &.current
+
+		&.next,
```
Let's remove empty blocks.

coreymcollins commented on assets/sass/plugins/_gravity-forms.scss:
```diff
@@ -166,9 +166,18 @@ $gf-input-padding: rem(6);
+		&::before {
+			border-color: $color-silver transparent transparent;
+			border-style: solid;
+			border-width: 12px (12px / 2) 0;
```
Do we need to do the math here? Could we just write `6px` instead since we're not doing math on a variable or anything like that?

jomurgel replied:
@coreymcollins I haven't touched any of the gravity form styles other than moving things around and updating some variables. I guess I'm not sure what reason this was done and/or if it's a requirement. Making a note here before making the update.

coreymcollins commented on assets/sass/structure/_layout.scss:
```diff
@@ -3,85 +3,144 @@
-	.primary {
-		margin-left: 0;
+	.grid-wrapper {
+		@include margin($gutter * 2 auto );
```
Remove extra space after `auto`

coreymcollins commented on assets/sass/utilities/mixins/_pxtoem.scss:
```diff
@@ -0,0 +1,19 @@
+// PX to EM
+//--------------------------------------------------------------
+
+// Convert pixels to ems
```
Can you add an example like you have in the other mixin files?

coreymcollins commented on assets/sass/utilities/mixins/_pxtorem.scss:
```diff
@@ -0,0 +1,20 @@
+// PX to Rem
+//--------------------------------------------------------------
+
+// Convert pixels to rems
```
Can you add an example like you have in the other mixin files?

coreymcollins commented on buddypress.php:
```diff
@@ -12,19 +12,17 @@
 
-	<div class="primary content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main" role="main">
```
Remove role.

coreymcollins commented on inc/template-tags.php:
```diff
@@ -496,10 +497,16 @@ function _s_display_numeric_pagination( $args = array() ) {
 	?>
-	<nav class="pagination-container">
+
+	<nav class="pagination-container container">
```
We should add aria labels here to specify what the nav is (something like `aria-label="<?php esc_html_e( 'Numeric pagination', 'swe' ); ?>"`.

coreymcollins commented on index.php:
```diff
@@ -14,8 +14,7 @@
 
-	<div class="primary content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main" role="main">
```
Remove role.

coreymcollins commented on page.php:
```diff
@@ -14,24 +14,22 @@
 
-	<div class="primary content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main" role="main">
```
Remove role.

coreymcollins commented on search.php:
```diff
@@ -9,8 +9,7 @@
 
-	<section class="primary content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main" role="main">
```
Remove role.

coreymcollins commented on single.php:
```diff
@@ -9,28 +9,28 @@
-	<div class="primary content-area col-l-8">
-		<main id="main" class="site-main">
+	<div class="display-flex grid-wrapper">
+		<main id="main" class="site-main left-two-thirds" role="main">
```
Remove role.

coreymcollins commented on template-acf.php:
```diff
@@ -11,8 +11,8 @@
 
-	<div class="content-area">
-		<main id="main" class="site-main">
+	<main id="main" class="site-main acf-content-blocks" role="main">
```
Remove role.

coreymcollins commented on template-parts/content-blocks/block-accordion.php:
```diff
@@ -13,15 +13,13 @@
 ?>
-	<div class="grid-x<?php echo esc_attr( $animation_class ); ?>">
-		<div class="cell">
+	<div class="<?php echo esc_attr( $animation_class ); ?>">
```
There should be a fallback if no animation class is set, otherwise we'll get an empty `class` attribute which will throw a11y warnings.

jomurgel replied:
Opted to set a default class inside the function rather than rewriting output in all of the template parts

coreymcollins commented on template-parts/content-blocks/block-accordion.php:
```diff
@@ -32,31 +30,31 @@
+								</button>
+							</div><!-- .accordion-item-header-->
+							<div id="<?php echo esc_attr( $item_content_id ); ?>" class="accordion-item-content" aria-hidden="true">
+								<?php echo wp_kses_post( $item_content ); ?>
```
Should we run this through the new `s_get_the_content` template tag instead for consistency?

coreymcollins commented on template-parts/content-blocks/block-accordion.php:
```diff
@@ -13,15 +13,13 @@
-	)
-);
+_s_display_block_options( array(
+	'container' => 'section', // Any HTML5 container: section, div, etc...
```
Based on conversations we've had around section and div usage, should we make this a div instead?

jomurgel replied:
@coreymcollins we might want to discuss this further. I think this would sort of justfiy having some function + acf field to determine the output. I could argue for div or even an aside or article depending on the content.

jomurgel replied:
The only one, based on intended use, that would for sure be a div or aside would be the cta block.

coreymcollins replied:
I figured this would be one we could all chat about, so I added an agenda item for today's scrum. I'm sure we'll all come to a decision on what would be best for us + users.

coreymcollins commented on template-parts/content-blocks/block-related_posts.php:
```diff
@@ -17,41 +17,43 @@
+			<?php endif; ?>
+		</div>
+
+		<div class="container display-flex <?php echo esc_attr( $animation_class ); ?>">
```
If no animation class is present, we'll wind up with an empty space here which will throw an a11y warning.

coreymcollins commented on template-parts/content-password-protected.php:
```diff
@@ -8,7 +8,7 @@
 
 ?>
-<div class="grid-x">
+<div class="">
```
Let's remove the empty class attribute.

coreymcollins commented on template-scaffolding.php:
```diff
@@ -11,10 +11,10 @@
-			<?php do_action( '_s_scaffolding_content' ); ?>
-		</main><!-- #main -->
-	</div><!-- .primary -->
+	<main id="main" class="site-main container" role="main">
```
Remove role.

coreymcollins commented on template-sidebar-left.php:
```diff
@@ -0,0 +1,37 @@
+
+		<?php get_sidebar(); ?>
+
+		<main id="main" class="site-main" role="main">
```
Remove role.

coreymcollins commented on template-sidebar-right.php:
```diff
@@ -11,8 +11,8 @@
-	<div class="primary content-area col-l-8">
-		<main id="main" class="site-main">
+	<div class="display-flex grid-wrapper">
+		<main id="main" class="site-main left-two-thirds" role="main">
```
Remove role.

coreymcollins reviewed (changes requested):
I've run through and reviewed the actual code changes, leaving a handful of comments here and there.

Before you get started on anything, I'm also going to run some visual QA tests (and try to break things!) as well as linting tests, so I'll add those as additional comments once I'm wrapped up with those.

coreymcollins commented:
I’m leaving visual notes on the theme in general. Some of these items are going to be due to grid changes and some may be issues present already. In the case of the latter, fixing those issues as part of this PR would be super helpful!

# Visual Notes

## Header
![](https://dl.dropbox.com/s/7v6zzj32bg4wnob/Screenshot%202019-01-16%2013.18.12.jpg?dl=0)
The search form no longer extends fully when opened

## 404
![](https://dl.dropbox.com/s/w4ft21mfq1sboyv/Screenshot%202019-01-16%2011.08.54.jpg?dl=0)
From mid-size down, the content runs into the edge of the screen. This may be a global issue but this is the first page I’m checking, so maybe not! Either way, we’ll want to make sure pages have left and right margin/padding to avoid the collision.

## Grid
![](https://dl.dropbox.com/s/1ib5wgzse0t5hqb/Screenshot%202019-01-16%2013.16.32.jpg?dl=0)
When sizing down the window, the content runs into the edge of the screen until we hit the 1170px breakpoint where the container gets left and right padding. You can see some edge collisions in the header as well.

## ACF Blocks
![](https://dl.dropbox.com/s/cnsaolo4nedfyu3/Screenshot%202019-01-16%2013.19.11.jpg?dl=0)
When a block has a background and is `container-width`, it needs some padding so the content doesn't hit the edge. This is not an issue below 1170px.

## Mobile Nav
![](https://dl.dropbox.com/s/8v33oip62hd781l/Screenshot%202019-01-16%2013.21.59.jpg?dl=0)
The link color is hard to read on the black background.

## Comments Mobile
![](https://dl.dropbox.com/s/uj26246y62hw9cs/Screenshot%202019-01-16%2013.23.50.jpg?dl=0)
When a lot of comments are replies, they smush up real good on small screens.

![](https://dl.dropbox.com/s/iygaggd4orheq1s/Screenshot%202019-01-16%2013.24.47.jpg?dl=0)
![](https://dl.dropbox.com/s/5hzjr080o5rilaa/Screenshot%202019-01-16%2013.24.24.jpg?dl=0)
The comment section and form run into the edges of the screen on mobile.

## Buttons
![](https://dl.dropbox.com/s/xz678fq02tp8t5j/Screenshot%202019-01-16%2013.25.16.jpg?dl=0)
![](https://dl.dropbox.com/s/tw476lbe676bxn0/Screenshot%202019-01-16%2013.28.27.jpg?dl=0)
Button hover/active states are super hard to read.

# Accessibility

## WCAG & 508
[a11y.zip](https://github.com/WebDevStudios/wd_s/files/2765948/a11y.zip)
Files attached here with some issues. Some of these are already noted in the code review, so just crosscheck to make sure we’re not missing anything.

I ran this on just the ACF Blocks page for now. 
I’d like to run this again once we get `master` merged back in as there have been a bunch of additions to the Scaffolding template which would be good to examine.

# CBT
Be sure to run CBT on main pages – the ACF Blocks page, the Scaffolding page (after merging in master), single posts, etc. Basically, let’s make sure any unique page is looking good.

# Merge `master`
Before passing this back for review, can you merge `master` into your branch? We’ve merged in the Sanitize and Scaffolding PRs and there are some conflicts that should be cleaned up + seeing how the new scaffolding elements look with your changes.

Thanks for diving so hard into this! This is looking really great, and creating a grid is pretty dang easy. I think one thing we’ll want to keep in mind is to communicate with our designers (i.e., you and @bezierer) to make sure we’re providing mockups that will be in line with our grid at whatever max content width we run at for each project. I wouldn’t want to provide mockups for a client with a grid that looks one way and then have to spend time customizing the grid in the theme to match super specific sizes if we can avoid that by keeping the mockups and CSS as closely tied to one another as possible.

jomurgel committed:
Merge branch 'master' into feature/#438-update-grid

jomurgel committed:
escape apostrophe on 404 page

jomurgel committed:
removes role="main" from <main> tag.

jomurgel committed:
removes clearfix and primary class tweak

jomurgel committed:
adds .hn class to headings loop

jomurgel committed:
removes empty style partials

jomurgel committed:
adds container-width qualifier for has-background padding

jomurgel committed:
escapes apostrophe in content-none partial

jomurgel committed:
rebuilds all

jomurgel committed:
removes empty selectors

jomurgel committed:
removes space

jomurgel committed:
removes empty ruleset

jomurgel committed:
adds example documentation to px to rem and px to em

jomurgel committed:
adds aria label to pgination container

jomurgel committed:
adds default not-animated class to animation class function

jomurgel committed:
run accordion content through _s_get_the_content function

ImBigWill reviewed (commented):
Rather than a screenshot, I just mentioned a small cleanup item similar to the 404 note, but for the content-none.php escaping of text. Overall this looks really good and I agree with Corey probably need to run through it after some of the updates since there are quite a few.

jomurgel committed:
removes empty class and indents correctly

jomurgel committed:
adds container to 404 content

jomurgel committed:
adds $container-offset breakpoint to accomodate breakpoint above container width for padding

jomurgel committed:
reduce container offset

jomurgel committed:
adds missing container classes

jomurgel committed:
updates mobile nav colors

jomurgel committed:
adds container class wrapper

jomurgel committed:
decreases padding on replies which helps mobile

jomurgel committed:
forces default color on buttons during hover/focus

jomurgel committed:
builds

jomurgel commented:
@coreymcollins a few followup questions.

Can you double check the search form again for me with some changes I've pushed up. This is what I see, but I also don't know if there's some part that is missing or some particular page that might be happening on, etc.

![screenshot 2019-01-17 14 19 37](https://user-images.githubusercontent.com/5230729/51349584-09cc9e00-1a63-11e9-9c92-6b3bba5d0846.jpg)

- Updated the 404 page
- Accommodated padding with the container class with an offset helper, breaking ~48px before the container to accommodate the padding offset.
- Updated the mobile menu styles to work better with the default WD_s colors.

![screenshot 2019-01-17 14 42 49](https://user-images.githubusercontent.com/5230729/51350726-2c13eb00-1a66-11e9-8bb3-0fb1e2797a5f.jpg)

- Decreased padding on replies to help out mobile a bit.

![screenshot 2019-01-17 15 02 46](https://user-images.githubusercontent.com/5230729/51351744-fa505380-1a68-11e9-8fd3-9f22dea59be6.jpg)

- Fixes container wrapping in comments prevents butting up against the edges.
- Resolves issue with button text readability on hover/focus.

![screenshot 2019-01-17 15 08 13](https://user-images.githubusercontent.com/5230729/51352050-bc076400-1a69-11e9-8478-b9f8ef63c7fb.jpg)

Several other visual items were addressed during code note updates.

Need to run through WCAG and 508 elements and CBT stuff yet. Master was merged back in, however, on this round of updates. Need to address mentioned `@extend` elements in the styles as well.

jomurgel committed:
resolves phpcs error

jomurgel committed:
replaces `@extend` with duplicate styles

jomurgel committed:
builds

jomurgel commented:
I've removed, in the post navigation, the use of `@extend` by duplicating the styles from `_layout.scss`. There's no filter to update `.posts-navigation` or `.post-navigation` so this was the cleanest way of doing it. Though, for the record, I don't think this is the best way to go about it either. I think using `@extend` here would have been perfectly fine.

Going through the WCAG and 508 errors now, though it appears that most of these are resolved at a glance.

Will make updates and hopefully be able to run CBT here shortly.

jomurgel committed:
adds id to carousel slides inside the `_s_display_block_options` function

jomurgel committed:
adds -image to card image alt tag to avoid duplicate link text error

jomurgel committed:
adds title field to video background

jomurgel committed:
adds conditional logic to video background title

jomurgel committed:
adds background video alt tag

jomurgel committed:
updates header nav

jomurgel committed:
removes padding from contaienr with background on mobile devices

jomurgel committed:
builds

jomurgel commented:
@coreymcollins I think I resolved the WCAG and 508 errors locally. Might try pulling down the branch and re-running your reports to confirm.

The only one I couldn't figure out what the `slick.css` being enqueued in the footer. I can't figure out why that's happening and/or how to move it if you have any ideas. You'd mentioned it potentially being done on purpose for Pagespeed Insights or some render-blocking resources issue, but I couldn't not see any reason that it would be enqueued in a place outside of the norm.

Went through and fixed a few warnings also, but most of those are non-issues, or were resolved by completely including image alt text in the admin or values where missing.

Ignored:
- [WCAG2 2.2.2 (A)] Videos that auto-play must have a means to stop them. (background videos)
- Whitespace warnings
- This warning
![screenshot 2019-01-18 09 54 21](https://user-images.githubusercontent.com/5230729/51401018-0b4fa200-1b07-11e9-8262-b8972e2dfefd.jpg)

Which must be an issue from older code since the nav has an aria label: https://github.com/WebDevStudios/wd_s/blob/feature/%23438-update-grid/inc/template-tags.php#L535

I ran CBT here, which is a little weird running from a local, but as far as I can tell things, in reality, look fine. 
- https://app.crossbrowsertesting.com/public/i55b132d134e1a40/screenshots/z4c8daa3e83745f97613

iPad and iPhone didn't work. I couldn't even get the live tests to work. Checking in the mobile chrome checker looks fine, those are here, but might need to find someone able to test an actual new device locally.
iPhone
![screencapture-192-168-1-237-3000-acf-blocks-2019-01-18-10_26_17](https://user-images.githubusercontent.com/5230729/51402733-8adf7000-1b0b-11e9-8669-9e56c0e69921.png)

iPad
![screencapture-192-168-1-237-3000-acf-blocks-2019-01-18-10_26_03](https://user-images.githubusercontent.com/5230729/51402738-8d41ca00-1b0b-11e9-9a6b-4476cd6e2c9d.png)

coreymcollins commented:
Just digging into this now and the first thing I wanted to test was the Slick Carousel stuff.

Right now we're enqueuing the scripts & styles in the template part for the carousel block. This is what puts the scripts and styles in the footer. If I enqueue these elsewhere, like in `scripts.php` they show up in the header.

Going to look at, instead of enqueuing the scripts directly in the template part, calling a function that enqueues them at the proper time and place. I'll update with findings!

coreymcollins committed:
Moves the enqueue of Slick scripts to avoid loading CSS in the footer incorrectly

coreymcollins commented:
This commit (https://github.com/WebDevStudios/wd_s/pull/447/commits/610ff58cedac3d86886b8a904325e36ee0f90c25) takes care of the issue of Slick CSS being enqueued in the footer.

JS still gets enqueued in the footer, as expected, but the CSS does not. We were calling `wp_enqueue_style` and `wp_enqueue_script` directly in the carousel content block template part which is what was causing the issue. I moved those enqueues to their own function (`_s_enqueue_slick_scripts`) and am calling that function in the carousel template part instead.

coreymcollins commented:
The search in the header is still causing an issue for me and it's kind of weird. Up until 1800px, I can see it expanding:
![](https://dl.dropbox.com/s/qki6fhl7n3gwl0u/Screenshot%202019-01-18%2013.17.40.jpg?dl=0)

Once I get past 1800px, it becomes the crunched up little dude seen here:
![](https://dl.dropbox.com/s/jz17c8czj0m8baa/Screenshot%202019-01-18%2013.18.32.jpg?dl=0)

Still running through the rest of the updates but wanted to log that one first since you had asked about it specifically.

coreymcollins committed:
Adjusts some indentations

coreymcollins committed:
Adds a fallback for video background if no description is set to avoid an empty space

coreymcollins committed:
Adjusts tags for video backgrounds

coreymcollins committed:
Fixes an output issue with footer social links

coreymcollins committed:
Rearranges accordion buttons to fix some a11y issues

coreymcollins committed:
Fixes a spacing issue

coreymcollins committed:
Adds a play/pause button so our background videos are more accessible

coreymcollins committed:
Compiles

coreymcollins committed:
Adjusts button screen reader text

coreymcollins commented:
I pushed up a handful of commits to take care of some a11y things, including adding a play/pause button to our blocks when a background video is used. This allows users who may want or need to pause a background video to do so. TVP still throws a warning around this as we're not using the native video controls but a JS solution with a button, but it is accessible/tab-able and includes screen reader text for an added bonus.

With your note on the `<nav>` item – this looks to be thrown because the primary nav doesn't have an aria label set. This is because we're using the core `wp_nav_menu` function and I don't think there's a way to add aria labels. We may be able to filter the output or maybe add a `<nav>` in the markup with the labels we need and change the way we output the menu?

Aside from the items noted above, I am seeing one last screen collision item noted below. Otherwise, this is looking really great!

I know Will looked it over and was happy with it, and I'd love to see if any other FEEs may want to give it a test to catch anything we may have missed (if anything) early next week. I don't want to delay it too long, though, as this is a big and good update that will be awesome to start using.

### Screen Collisions
The edge-of-screen collisions are looking pretty good for the most part, but the nav in the header still runs into the edge around 1170px as seen here:
![](https://dl.dropbox.com/s/fdm3zge8ztgsf5y/Screenshot%202019-01-18%2013.21.05.jpg?dl=0)

It looks like around 1194px that things jump back into place:
![](https://dl.dropbox.com/s/xu1frni2kp8viu2/Screenshot%202019-01-18%2013.22.00.jpg?dl=0)

coreymcollins committed:
Adds a check to see if an image exists. Sometimes it may not!

jomurgel committed:
restructures main nav wp_nav_menu and adds aria label

jomurgel commented:
Pushed up another commit that adds the aria-label to the main navigation mentioned above.

``` php
		<nav id="site-navigation" class="main-navigation" aria-label="<?php esc_html_e( 'Main Navigation', '_s' ); ?>">
			<?php
			wp_nav_menu( array(
				'fallback_cb'     => false,
				'theme_location'  => 'primary',
				'menu_id'         => 'primary-menu',
				'menu_class'      => 'menu dropdown',
				'container'       => false,
			) );
			?>
		</nav><!-- #site-navigation-->
```

jomurgel committed:
adds container class to nav wrapper

jomurgel committed:
cleans up styles

jomurgel committed:
adds container to nav menu ul

jomurgel committed:
resolves padding issues on desktop devices for main navigation

jomurgel committed:
removes unecessary acf-content-blocks class if password protected page is password protected

jomurgel committed:
cleanup

jomurgel committed:
builds

jomurgel committed:
Merge branch 'master' into feature/#438-update-grid

jomurgel commented:
Pushed a couple additional updates to fix the nav issue above, and brought the branch current with master. 👍

coreymcollins commented:
@jomurgel is this ready for review again? I see your updates above but it's still assigned to you with the `returned` label, so I wasn't sure where it stood at this point. Thanks!

coreymcollins committed:
Adds a hidden h1 post title to the ACF page template for screen readers

jomurgel committed:
resolves issues with searchform width on larger screens

jomurgel committed:
fixes calc issue with variable

jomurgel committed:
builds

jomurgel commented:
@coreymcollins updated a couple things here:
- Resolved issues with the width of the search form shown in your screenshots above. Removed any real width restriction above desktop devices.
- Also resolved an issue with the use of `calc` and variables in the grid mixins which was just breaking the value of `flex-basis`.

gregrickaby reviewed (changes requested):
Hey @jomurgel and @coreymcollins! Great work here. I really appreciate dropping the sidebar, and all other the little details. Here are a couple of suggestions:

1. Move _Font Color_ and _Animation_ from "Other Options" --> "Display Options".
2. When I set a block to Full [width] in Display Options, the background doesn't actually extend full width. It could be, that I misunderstand what full width should be?

coreymcollins commented:
Thanks, Greg! Interesting catch on no. 2 above. That definitely was extending full-width previously (per the screenshots earlier in the PR), so there must be a commit here that walked that back at some point.

I think the adjustment in no. 1 makes sense as well. 👍

gregrickaby commented:
@coreymcollins I even re-ran Gulp just to make sure. It's not full-width:

![screenshot](https://dl.dropbox.com/s/wdt21qnsk7mutkf/Screen%20Shot%202019-01-24%20at%2010.53.30.png?dl=0)

coreymcollins commented:
Yup, looks like this commit is the culprit by adding `container` as a class: https://github.com/WebDevStudios/wd_s/pull/447/commits/482d0938b1e01b18e1d1556e3719d39f2c9646cd

I'll get with Jo to see if adding that class was intended or if we can remove it without issue.

jomurgel committed:
removes container class from <main> and moves to ternary in acf template

coreymcollins committed:
Fixes an extra space in the ACF template

coreymcollins committed:
Fixes a space issue in the carousel class

coreymcollins committed:
Adjusts carousel markup and class names to pull in color and animation

coreymcollins committed:
Moves colors and animations to Display Options instead of Other Options

coreymcollins commented:
Pushed some updates here to:
- Move Animation & Font Color to Display Options (https://github.com/WebDevStudios/wd_s/pull/447/commits/733fde739b4cb9903b59a9d51dbb4182001e64cc)
- Update the ACF functions to look for those options in their new spots (https://github.com/WebDevStudios/wd_s/pull/447/commits/733fde739b4cb9903b59a9d51dbb4182001e64cc)
![](https://dl.dropbox.com/s/3aqokg1vmqvea00/Screenshot%202019-01-24%2011.43.45.jpg?dl=0)
- Update the carousel module to accept the animation and color options in its parent container (https://github.com/WebDevStudios/wd_s/pull/447/commits/196b2be2a6a9e705d52ef12abcb99ab0eb080575). Any attempt at trying to grab a gif of this failed miserably with a super choppy gif 😆 
- Fix a couple of space issues (https://github.com/WebDevStudios/wd_s/pull/447/commits/7465d707e30870e7608b523dff7a38f8f32feae9, https://github.com/WebDevStudios/wd_s/pull/447/commits/9a5937d5fa752a54c4e0578dee77487f803f9584)

gregrickaby commented:
@coreymcollins Awesome! I think we can do it even better. Background Options are technically part of the front-end display, and should probably be consolidated to Display Options.

coreymcollins commented:
I like this idea in theory, but one sticky thing is that we don't offer Background Options on the Carousel since each slide has its own options for setting the slide content & background.

If we move Background Options into Display Options, we run into a problem where they display in that tab but won't actually be used for the Carousel block. I don't think there's a way to conditionally hide a set of fields in a tab group based on the content block type unless we throw some JS in there.

Going to think through it some more and see if there's a way to move them and still elegantly hide them for blocks that won't use Background Options.

coreymcollins commented:
I think the best call may be to keep Background Options as-is so we can keep them off of the Carousel block (and any other blocks that may not need/be able to use background options) and merge this puppy in if everything else is looking good from your review.

We can always explore sneaky ways to manipulate tabs based on flexible content layouts further, but I'd love to get the grid stuff updated and merged in today if at all possible. Let me know what you think when you have a chance!

coreymcollins committed:
Removes references to Bourbon and Neat throughout various files

coreymcollins committed:
Updates Yarn file

gregrickaby commented:
@coreymcollins Sounds good. Thanks for taking a look!
